File: services/processors/deposito_processor.py

from interfaces.i_transacao_processor import ITransacaoProcessor

class DepositoProcessor(ITransacaoProcessor):
    def __init__(self, conta_repository, logger):
        self.conta_repository = conta_repository
        self.logger = logger

    def processar(self, transacao):
        conta_saldo_destino = self.conta_repository.get_saldo(transacao.conta_destino)
        
        if not conta_saldo_destino:
            self.logger.log(f"Conta destino {transacao.conta_destino} não encontrada")
            return False

        conta_saldo_destino.saldo += transacao.valor
        self.conta_repository.atualizar_saldo(conta_saldo_destino)

        self.logger.log(f"Depósito numero {transacao.correlation_id} foi efetivado com sucesso! Novo saldo: {conta_saldo_destino.saldo}")
        return True

File: services/processors/DepositoProcessor.js

const ITransacaoProcessor = require('../../interfaces/ITransacaoProcessor');

class DepositoProcessor extends ITransacaoProcessor {
    constructor(contaRepository, logger) {
        super();
        this.contaRepository = contaRepository;
        this.logger = logger;
    }

    processar(transacao) {
        const contaDestino = this.contaRepository.getSaldo(transacao.contaDestino);
        
        if (!contaDestino) {
            this.logger.log(`Conta ${transacao.contaDestino} não encontrada para depósito`);
            return false;
        }

        contaDestino.saldo += transacao.valor;
        this.contaRepository.atualizarSaldo(contaDestino);

        this.logger.log(`Depósito ${transacao.correlationId} processado. Novo saldo: ${contaDestino.saldo}`);
        return true;
    }
}

module.exports = DepositoProcessor;
